### PR TITLE
Rewrite Applications view

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Rewrite Applications to React & Pamela

**Related issue(s)**

### CODE REVIEW

File: core-ui/src/components/Predefined/Details/Application/ConnectApplicationModal.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  const [url, setUrl] = React.useState();
+
+  const postRequest = usePost();
+  const deleteTokenRequest = useDelete();
+  const getRequest = useSingleGet();
+
+  async function performUrlFetch() {
+    setUrl('Loading...');
+
+    const defaultNamespace = 'default'; // just use the default namespace
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/Details/Application/ConnectApplicationModal.js:36]: why only default?

File: core-ui/src/components/Predefined/List/Application/Applications.list.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import CreateApplicationModal from './CreateApplicationModal';
+
+export const ApplicationsList = DefaultRenderer => ({ ...otherParams }) => {
+  const customHeaderActions = <CreateApplicationModal />;
+  //   const customColumns = [
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/List/Application/Applications.list.js:6]: :)

File: core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        <TextFormItem
+          inputKey="description"
+          label="Description"
+          placeholder="Specify a description for your Application"
+          onChange={e =>
+            setApplication({ ...application, description: e.target.value })
+          }
+        />
+        <LabelSelectorInput
+          showHelp={false}
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js:76]: why won't we show info?

**Review by dariadomagala — COMMENTED** [review]
Please check comments

File: core-ui/src/components/Predefined/Details/Application/ConnectApplicationModal.js (hunk 1 above)
Comment (Wawrzyn321) [reply core-ui/src/components/Predefined/Details/Application/ConnectApplicationModal.js:36]: `TokenRequests` is namespaceable, so we need a namespace here.
It's a similar procedure to the one performed in step 3 of [docs](https://kyma-project.io/docs/root/getting-started/#getting-started-connect-an-external-application-create-the-application-and-retrieve-a-token).

File: core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  const postRequest = usePost();
+
+  async function handleFormSubmit() {
+    try {
+      const applicationInput = createApplicationInput(application);
+      await postRequest(
+        '/apis/applicationconnector.kyma-project.io/v1alpha1/applications',
+        applicationInput,
+      );
+      onCompleted(application.name, `Application created succesfully`);
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js:47]: I'm afraid it won't be possible, as data is fetched on the same level that the header (containing the `CreateApplicationModal`) is rendered. See `ResourcesList` component.

File: core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    } catch (e) {
+      console.warn(e);
+      onError('Cannot create application', e.message);
+    }
+  }
+
+  return (
+    <form onChange={onChange} ref={formElementRef} onSubmit={handleFormSubmit}>
+      <FormSet>
+        <TextFormItem
```
Comment (dariadomagala) [inline core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js:57]: we have component called `k8sNameInput` or something similar. could we use it here or is the regex/logic any different?

Comment (Wawrzyn321) [reply core-ui/src/components/Predefined/List/Application/CreateApplicationModal.js:57]: Well, on CBS side regex part looked like this:
```
// This regex comes from the k8s resource name validation and has been checked against traversal attack
// https://github.com/kubernetes/kubernetes/blob/v1.10.1/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L126
appNameRegex      = `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`
```

**Review by dariadomagala — DISMISSED** [review]

**Review by dariadomagala — APPROVED** [review]

**Pull request merged**